src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -154,19 +154,19 @@ internal static bool DoSslHandshake(SafeSslHandle context, byte[] recvBuf, int r
             return stateOk;
         }
 
-        internal static int Encrypt(SafeSslHandle context, byte[] buffer, int offset, int count, out Ssl.SslErrorCode errorCode)
+        internal static int Encrypt(SafeSslHandle context, byte[] input, int offset, int count, ref byte[] output, out Ssl.SslErrorCode errorCode)
         {
-            Debug.Assert(buffer != null);
+            Debug.Assert(input != null);
             Debug.Assert(offset >= 0);
             Debug.Assert(count >= 0);
-            Debug.Assert(buffer.Length >= offset + count);
+            Debug.Assert(input.Length >= offset + count);
```

Comment:
Well, I went with `input.Length - offset >= count`, after verifying that `offset <= input.Length`.  But, yeah, good catch.